### DISCUSSION TITLE
add friend

### DISCUSSION THREAD

**Discussion by tom-jm69** [ValvePython/steam#398] [Q&A] [answered]
hey, how do i add/accept a friend request? thats my code so far, i already found that (https://steam.readthedocs.io/en/stable/api/steam.client.builtins.html?highlight=add#steam.client.builtins.friends.SteamFriendlist.add), but i don't know on how to use the add

```python
@sc.on(EMsg.ClientFriendsList)
def friend_add(msg):
    steamid = msg.body.friends[0].ulfriendid
    #steam client = sc

**Comment by everlyy** [comment, marked as answer]
You can use the SteamFriendlist to manage everything friend stuff related. If you want to add someone after they've sent a friend request you can do something like
```py
from steam.client.builtins.friends import SteamFriendlist

friendslist = SteamFriendlist(sc) # sc is your SteamClient

@friendslist.on("friend_invite")
def fl_friend_invite(user):
	friendslist.add(user)
```

Hope this helps!!

**Reply by tom-jm69** [reply]
thank you very much!
